test/controllers/IllnessStartDateController.test.ts

```diff
@@ -0,0 +1,156 @@
+                        expect(response.text).to.include(pageHeading)
+                            .and.to.include(invalidStartDayErrorMessage)
+                            .and.to.not.include(invalidDateErrorMessage)
+                            .and.to.not.include(invalidDateFutureErrorMessage);
+                    });
+            });
+        it('should return 422 response with rendered error message invalid start date (all zeros) was submitted',
+            async () => {
+                const app = createApp();
+                await request(app).post(ILLNESS_START_DATE_PAGE_URI)
+                    .send({startDay: '00', startMonth: '00', startYear: '00'})
+                    .expect(response => {
+                        expect(response.status).to.be.equal(UNPROCESSABLE_ENTITY);
+                        expect(response.text).to.include(pageHeading)
+                            .and.to.include(errorSummaryHeading)
+                            .and.to.include(invalidDateErrorMessage);
+                    });
+            });
+        it('should return 422 response with rendered error message invalid start date (non-existing) was submitted',
+            async () => {
+                const app = createApp();
+                await request(app).post(ILLNESS_START_DATE_PAGE_URI)
+                    .send({startDay: '32', startMonth: '13', startYear: '2020'})
+                    .expect(response => {
+                        expect(response.status).to.be.equal(UNPROCESSABLE_ENTITY);
+                        expect(response.text).to.include(pageHeading)
+                            .and.to.include(errorSummaryHeading)
+                            .and.to.include(invalidDateErrorMessage);
+                    });
+            });
+        it('should return 422 response with rendered error message invalid start date (in future) was submitted',
+            async () => {
+                const app = createApp();
+                await request(app).post(ILLNESS_START_DATE_PAGE_URI)
+                    .send({startDay: '01', startMonth: '01', startYear: '2029'})
+                    .expect(response => {
+                        expect(response.status).to.be.equal(UNPROCESSABLE_ENTITY);
+                        expect(response.text).to.include(pageHeading)
+                            .and.to.include(errorSummaryHeading)
+                            .and.to.include(invalidDateFutureErrorMessage);
+                    });
+            });
+        it('should redirect to entry page when illness reason feature is disabled', async () => {
+
+            process.env.ILLNESS_REASON_FEATURE = '0';
+
+            const app = createApp();
+            await request(app).post(ILLNESS_START_DATE_PAGE_URI)
+                .send({})
+                .expect(response => {
+                    expect(response.status).to.be.equal(MOVED_TEMPORARILY);
+                    expect(response.header.location).to.include(ENTRY_PAGE_URI);
+                });
```

Comment:
isnt this already tested?

Reply:
i've removed the these duplicate controller integration tests from the middleware test